src/test/java/nl/ordina/jobcrawler/service/SkillServiceTest.java

```diff
@@ -0,0 +1,297 @@
+package nl.ordina.jobcrawler.service;
+
+import nl.ordina.jobcrawler.model.Skill;
+import nl.ordina.jobcrawler.model.Vacancy;
+import nl.ordina.jobcrawler.repo.SkillRepository;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+
+class SkillServiceTest {
+
+    static private SkillRepository skillRepositoryMock;
+    static private SkillService skillService;
+
+    private Vacancy vacancy;
+    private List<Vacancy> vacanciesDB;  // fake "database" for vacancy
+    private String skillName;
+    private Skill skill;
+    private List<Skill> skillDB;    // fake "database" for skill
+
+    @BeforeAll
+    static void beforeAll() {
+        skillRepositoryMock = Mockito.mock(SkillRepository.class);
+        skillService = new SkillService(skillRepositoryMock);
+    }
+
+    @BeforeEach
+    void init() {
+
+        vacancy = Vacancy.builder() // making a vacancy
+                .id(UUID.randomUUID())
+                .vacancyURL("example.com")
+                .title("job title Example 1")
+                .broker("broker Example")
+                .vacancyNumber("1")
+                .hours("30")
+                .location("location example")
+                .postingDate("14 April 2020")
+                .about("this is a description of the example job")
+                .skills(new HashSet<Skill>())
+                .build();
+
+        skillName = "example";
+        skill = Skill.builder()
+                .id(UUID.randomUUID())
+                .name(skillName)
+                .vacancies(new HashSet<Vacancy>())
+                .build();
+
+        vacancy.addSkill(skill);    // making the link between the vacancy and the skill
+
+        skillDB = new ArrayList<Skill>();
+        vacanciesDB = new ArrayList<Vacancy>();
+
+        vacanciesDB.add(vacancy);   // putting the vacancy in the fake vacancy "database"
+        skillDB.add(skill); // putting the vacancy in the fake skill "database"
+    }
+
+    Skill skillFactory(String skillName){
+        return Skill.builder()
+                .id(UUID.randomUUID())
+                .name(skillName)
+                .vacancies(new HashSet<Vacancy>())
+                .build();
+    }
+
+    @Test
+    void getAllSkills() {
+        /* uses from repository methods: findAll */
+
+        Mockito.when(skillRepositoryMock.findAll()).thenReturn(skillDB);
+        assertThat(skillService.getAllSkills()).isEqualTo(skillDB);
+    }
```

Comment:
Not sure if needed, but maybe verify that `skillRepositoryMock.findAll()` was called only once using the `skillService.getAllSkills()` method?

```java
verify(skillRepositoryMock, times(1)).findAll();
```

Reply:
good idea

Reply:
However I am not sure but this would be more for integration test, I thought.

Reply:
Check [TDD with Spring and JUnit](https://app.pluralsight.com/courses/2e2fc36b-38aa-4634-b2df-a7661f15613b/table-of-contents) 3rd module 5th video at 4:25. 
The person first makes a list of 2 products which will be returned by the mock repo. Then the call to the service is made, and the assert is made on the returned list. It does seem kinda redundant because they could check on the method call directly, that is why I am not sure if this is the best way.